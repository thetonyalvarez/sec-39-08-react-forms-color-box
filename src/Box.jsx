import { useState } from 'react'
import './Box.css'
import { Box as BoxUI } from '@mui/material'
import Button from '@mui/material/Button'

const Box = ({ id, bgColor = 'red', width = 100, height = 100, handleRemove }) => {

  const remove = () => handleRemove(id)

  return (
    <BoxUI
      className="Box"
      sx={{
        backgroundColor: bgColor,
        width: width + 'px',
        height: height + 'px'
      }}
    >
      <Button onClick={remove}>X</Button>
    </BoxUI>
  )
}

export default Box;