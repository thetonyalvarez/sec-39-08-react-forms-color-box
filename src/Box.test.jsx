import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import Box from './Box'

afterEach(cleanup)

// smoke test
it("renders without crashing", function() {
  render(<Box />)
});

// snapshot
it("matches snapshot", function() {
  const {asFragment} = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

it("should render a blue box that is 50px x 50px", function() {
  const { getByTestId } = render(<Box id={1} bgColor={"blue"} width={50} height={50} />)

  const box = getByTestId('box')

  expect(box).toHaveStyle('background-color: blue')
  expect(box).toHaveStyle('width: 50px')
  expect(box).toHaveStyle('height: 50px')
})