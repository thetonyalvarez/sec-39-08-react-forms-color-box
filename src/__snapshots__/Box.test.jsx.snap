// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box MuiBox-root css-1ym0vmf"
    data-testid="box"
  >
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      X
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
